android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -28,6 +29,45 @@ open class MainActivity : ReactActivity() {
         }
     }
 
+    /* Returns true just if we did handle the intent. */
+    private fun maybeHandleIntent(
+        intent: Intent?,
+        contentResolver: ContentResolver
+    ): Boolean {
+        if (intent == null) {
+            return false
+        }
+
+
+        val host = (application as ReactApplication).reactNativeHost
+        val reactContext = host.tryGetReactInstanceManager()?.currentReactContext
```

Comment:
Hmm, actually I don't love having these lines run here, above the `when (intent.action)`.

The thing is that that means they run even if they're not needed -- if the intent is something else. They look like they could be cheap and side-effect-free… or could not be. I think there isn't any live case where it'd matter even if they are -- the only intents we handle other than the two cases below are the web-auth intents, where we'll want a React instance/context anyway -- but I'd rather not have to worry about that possibility coming up in the future.

Maybe simplest is I'll just add a commit at the end to address this.

Reply:
> Maybe simplest is I'll just add a commit at the end to address this.

OK, SGTM, thanks!